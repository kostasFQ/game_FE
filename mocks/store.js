export default {
  gameStarted: false,
  gameOver: false,
  totalCount: 0,
  initialTime: 15
}