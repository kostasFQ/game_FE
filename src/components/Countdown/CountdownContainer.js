import { connect } from 'react-redux';
import { toggleGameStarted } from 'store/game/actions';
import Countdown from './CountdownComponent';

const mapDispatchToProps = dispatch => ({
  toggleGameStarted: bool => dispatch(toggleGameStarted(bool)),
})

export default connect(null, mapDispatchToProps)(Countdown);