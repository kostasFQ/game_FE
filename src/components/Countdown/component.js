import React, { Component } from 'react';

class Countdown extends Component {
  state = {
    secondsToStart: 3,
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      this.setState(state => {
        return { secondsToStart: state.secondsToStart - 1 };
      });
    }, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  componentDidUpdate() {
    const { secondsToStart } = this.state;
    const { toggleGameStarted } = this.props;

    if (secondsToStart === 0) {
      clearInterval(this.interval);
      toggleGameStarted(true);
    }
  }

  render() {
    const { secondsToStart } = this.state;
    return (
      <div >start from {secondsToStart} seconds</div>
    );
  }
}

export default Countdown;