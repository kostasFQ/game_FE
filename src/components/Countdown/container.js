import { connect } from 'react-redux';
import { toggleGameStarted, toggleGameOver } from 'store/game/actions';
import Countdown from './component';

const mapDispatchToProps = dispatch => ({
  toggleGameStarted: bool => dispatch(toggleGameStarted(bool)),
  toggleGameOver: bool => dispatch(toggleGameOver(bool)),
})

export default connect(null, mapDispatchToProps)(Countdown);