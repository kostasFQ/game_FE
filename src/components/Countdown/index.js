import Countdown from './CountdownContainer';

export default Countdown;