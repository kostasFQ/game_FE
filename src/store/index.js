import { combineReducers, createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import game from './game/reducer';

const store = combineReducers({
  game
});

export default createStore(store, compose(applyMiddleware(thunk), window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
