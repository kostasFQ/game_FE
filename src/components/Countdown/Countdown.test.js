import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { StaticRouter } from 'react-router-dom';
import store from '../../../mocks/store';

import Countdown from './container';

const initialState = { store };
const mStore = configureMockStore([thunk]);
const mockStore = mStore(initialState);


test('Countdown should be render', () => {
  const instance = renderer.create(
    <Provider store={mockStore}>
      <StaticRouter location="/" context={{}}>
        <Countdown />
      </StaticRouter>
    </Provider>,
  );

  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})