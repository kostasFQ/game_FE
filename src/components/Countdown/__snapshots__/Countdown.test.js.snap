// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown should be render 1`] = `
<div>
  start from 
  3
   seconds
</div>
`;
